packages/realm-react/src/cachedCollection.ts

```diff
@@ -196,6 +197,9 @@ export function createCachedCollection<T extends Realm.Object<any>>({
 
   const tearDown = () => {
     if (!isDerived) {
+      if (setImmediateId) {
+        clearImmediate(setImmediateId);
```

Comment:
@bimusiek, please also `declare` the `clearImmediate` function in [runtime.d.ts](https://github.com/realm/realm-js/blob/main/packages/realm-react/src/runtime.d.ts), similarly to what we have for the [realm package](https://github.com/realm/realm-js/blob/fac140a239ac4df7e762e0427ad23c34d6520f14/packages/realm/src/runtime.d.ts#L22).

Also wondering if resetting the id here to `undefined` once cleared would be good.

Reply:
Declaration added & clearing setImmediateId.